pkg/capabilities/pb/capabilities.proto

```diff
@@ -49,6 +49,7 @@ message UnregisterTriggerRequest {
 service TriggerExecutable {
   rpc RegisterTrigger (RegisterTriggerRequest) returns (google.protobuf.Empty) {}
   rpc UnregisterTrigger (UnregisterTriggerRequest) returns (google.protobuf.Empty) {}
+  rpc GetRequestConfigJsonSchema(google.protobuf.Empty) returns (CapabilityResponse) {}
```

Comment:
Hmmm, I was thinking that this would be something that's internal to the capability, i.e., when `RegisterTrigger` is called, it (internally) calls `ValidateConfig` and returns an error if config is invalid, otherwise it stores the CapabilitySpecific config internally.

Reply:
I was originally thinking what you are proposing here too, but what I realized is that by exposing this via endpoint we can provide a much better DX.

With this endpoint, we can do something like creating a full JSON schema that can be used to lint a YAML workflow file.

Reply:
I also want to internally call `ValidateConfig` during entrypoints like `RegisterTrigger`, then we get both:
1. The ability to provide workflow authors with a JSON schema meant for linting
2. Capabilities always validate their config and inputs upon execution